lambdas/utils/decorators/validate_document_type.py

```diff
@@ -0,0 +1,42 @@
+from typing import Callable
+
+from enums.supported_document_types import SupportedDocumentTypes
+from utils.lambda_response import ApiGatewayResponse
+
+
+def validate_document_type(lambda_func: Callable):
+    """A decorator for lambda handler.
+    Verify that the incoming event contains a valid document Type (ARF or LG)
+    If not, returns a 400 Bad request response before the lambda triggers.
+
+    Usage:
+    @validate_patient_id
+    def lambda_handler(event, context):
+        ...
+    """
+
+    def interceptor(event, context):
+        try:
+            doc_type = event["queryStringParameters"]["docType"]
+            if doc_type is None:
+                return ApiGatewayResponse(
+                    400, "docType not supplied", "GET"
+                ).create_api_gateway_response()
+            if not doc_type_is_valid(doc_type):
+                return ApiGatewayResponse(
+                    400, "Invalid document type requested", "GET"
+                ).create_api_gateway_response()
+        except KeyError as e:
+            return ApiGatewayResponse(
+                400, f"An error occurred due to missing key: {str(e)}", "GET"
+            ).create_api_gateway_response()
+
+        # Validation done. Return control flow to original lambda handler
+        return lambda_func(event, context)
+
+    return interceptor
+
+
+def doc_type_is_valid(doc_type: SupportedDocumentTypes) -> bool:
```

Comment:
I think it is not enough. What if we get 'sdfjfvsjhfvsukjARFfjdhtgdkjughLG'? it shouldn't pass validation.

Reply:
Because we wouldn't get that, the doctype isn't sent by the user so we should only be getting stuff that we're sending back based on what step of the UI is. I can totally do it if you insist but I think it's over-engineering it for what it needs to do right now.

Reply:
I know the query string are sent by our UI, but what is the point of this validation if not to make sure nobody passing our UI and trying to send queries directly to our Backend?